lib/util.py

```diff
@@ -62,18 +64,34 @@ def log_error(self, msg):
         self.logger.error(self.log_prefix + msg)
 
 
-# Method decorator.  To be used for calculations that will always
-# deliver the same result.  The method cannot take any arguments
-# and should be accessed as an attribute.
 class cachedproperty(object):
-
-    def __init__(self, f):
+    '''
+    Method decorator.
+    The result of the decorated method will be cached forever or for
+    `expire_time` seconds and will be available as an attribute with the same
+    name as the decorated method.
+    '''
+    def __init__(self, *args, expire_time=None):
```

Comment:
Can you create a new decorator entirely for time-cached properties?  I'd like to keep the existing one as-is.  Nice idea, b.t.w.